### PR TITLE
Logging decisions and service responses to Kibana and Splunk

### PR DESCRIPTION
**Pull request by bravesirrobin81**

### CODE REVIEW

File: app/uk/gov/hmrc/agentaccesscontrol/connectors/desapi/DesAgentClientApiConnector.scala [hunk 1]
```diff
@@ -19,26 +19,34 @@ package uk.gov.hmrc.agentaccesscontrol.connectors.desapi
...
+    httpGet.GET(url) map { r =>
+      logResponse(saUtr, agentCode, r.body)
+      foundResponseReads.reads(Json.parse(r.body)).get
+    } recover {
       case _: NotFoundException => NotFoundResponse
     }
+  }
+
+  def logResponse(saUtr: SaUtr, agentCode: AgentCode, body: String)(implicit hc: HeaderCarrier) = {
+    auditService.auditEvent(AgentAccessControlEvent.GGW_Response, agentCode, saUtr, Seq("body" -> body))
```
Comment (szmg) [inline app/uk/gov/hmrc/agentaccesscontrol/connectors/desapi/DesAgentClientApiConnector.scala]: Shouldn't it be `DES_Response`?

Comment (bravesirrobin81) [inline app/uk/gov/hmrc/agentaccesscontrol/connectors/desapi/DesAgentClientApiConnector.scala]: Yes.

Comment (bravesirrobin81) [inline app/uk/gov/hmrc/agentaccesscontrol/connectors/desapi/DesAgentClientApiConnector.scala]: Changed, and added the missing test.

**Comment by ZDevelop94** [comment]
does this piece of code audit a response from DES in splunk or another monitoring service? (Can only see lines 45-48).

**Comment by szmg** [comment]
@ZDevelop94 simple logging goes to ElasticSearch/Kibana, while `AuditService` sends events to Splunk.

We must not log secure data to the former. more info [here](https://confluence.tools.tax.service.gov.uk/display/TEC/2015/01/06/Logging+best+practices)

**Comment by ZDevelop94** [comment]
Thank you, sorry for the questions.

**Pull request merged**